### ISSUE TITLE
AGS 4 / New compiler: regular arrays of dynamic arrays

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2080]
I might have missed something, but it seems there's no way to have a **regular array** storing pointers to dynamic array inside.
I cannot think of reasons why this would not be possible (because regular arrays of pointers are supported), so likely it's a lack of script syntax.

Things I tried:
```
Character*[] arr1[10];
Character[] arr2[10];
Character* arr3[10][];
Character arr4[10][];
```
none of the above worked.

Hence my proposal is to support regular arrays of dynamic arrays. Honestly, I am unsure of what the correct syntax should be.

See Also: #1258

**Existing workaround**

The workaround is to make a struct (or managed struct) with dynamic array inside, although that creates an extra type and extra level of indirection:
```
struct Array {
    int a[];
};

Array a[10];
```

Unfortunately, dynamic arrays of dynamic arrays are also impossible at the moment, see #1258.

**Comment by fernewelten** [comment]
I think it should be possible. I'll look into this and see why the compiler doesn't like it.

**Comment by ivan-mogilko** [comment]
@fernewelten if you got more spare time afterwards, could you also look if #1818 (dynamic arrays of regular structs) is easy enough to accomplish? I imaging that in the best case this is a matter of checking the element kind (regular/managed), and choose between directly accessing array memory with offset, or dereferencing a managed pointer and accessing its element.

**Comment by fernewelten** [comment]
Will do!

**Comment by fernewelten** [comment]
Addressed in #2106.

**Comment by ivan-mogilko** [comment]
Implemented by #2106.

The syntax is e.g.:
```
int arr1[5][][];
Character* arr2[10][];
```
